### ISSUE TITLE
Missing Type Annotations

### ISSUE THREAD

**Issue by FoamyGuy** [adafruit/Adafruit_CircuitPython_BitbangIO#21]
There are missing type annotations for some functions in this library.

The `typing` module does not exist on CircuitPython devices so the import needs to be wrapped in try/except to catch the error for missing import. There is an example of how that is done here:

```python
try:
    from typing import List, Tuple
except ImportError:
    pass
```

Once imported the typing annotations for the argument type(s), and return type(s) can be added to the function signature. Here is an example of a function that has had this done already:

```python
def wrap_text_to_pixels(
    string: str, max_width: int, font=None, indent0: str = "", indent1: str = ""
) -> List[str]:
```
If you are new to Git or Github we have a guide about contributing to our projects here: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

There is also a guide that covers our CI utilities and how to run them locally to ensure they will pass in Github Actions here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code In particular the pages: `Sharing docs on ReadTheDocs` and `Check your code with pre-commit` contain the tools to install and commands to run locally to run the checks.

If you are attempting to resolve this issue and need help, you can post a comment on this issue and tag both @foamyguy and @kattni or reach out to us on Discord: https://adafru.it/discord in the `#circuitpython-dev` channel.

The following locations are reported by mypy to be missing type annotations:

- [ ] adafruit_bitbangio.py:65
- [ ] adafruit_bitbangio.py:79
- [ ] adafruit_bitbangio.py:117
- [ ] adafruit_bitbangio.py:124
- [ ] adafruit_bitbangio.py:134
- [ ] adafruit_bitbangio.py:199
- [ ] adafruit_bitbangio.py:225
- [ ] adafruit_bitbangio.py:249
- [ ] adafruit_bitbangio.py:255
- [ ] adafruit_bitbangio.py:264
- [ ] adafruit_bitbangio.py:278
- [ ] adafruit_bitbangio.py:315
- [ ] adafruit_bitbangio.py:334
- [ ] adafruit_bitbangio.py:340
- [ ] adafruit_bitbangio.py:372
- [ ] adafruit_bitbangio.py:418

**Comment by tcfranks** [comment]
ok......question about buffer types, ie:
def writeto_then_readfrom(
        self,
        address,
        buffer_out,
        buffer_in,
        *,
        out_start=0,
        out_end=None,
        in_start=0,
        in_end=None

I suspect buffer_in and buffer_out are bytearrays, but how to annotate them?

**Comment by dhalbert** [comment]
`ReadableBuffer` and `WriteableBuffer` would make sense here, I think. These are in `circuitpython_typing`.

**Comment by tcfranks** [comment]
y'all have a position on f-strings?
Pylint
adafruit_bitbangio.py:282:31: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
actual line:
            raise RuntimeError("Device not responding at 0x{:02X}".format(address))

**Comment by kattni** [comment]
@tcfranks Please feel free to use them!

**Comment by tcfranks** [comment]
great, dropped the setup file, but still don't know how to interpret the error that was just thrown....

**Comment by tcfranks** [comment]
> 

all pylint locally complains about is pylint directives (in the original file) and line endings....

**Comment by tcfranks** [comment]
is this closed, then?

**Comment by tcfranks** [comment]
ok, a question I haven't come across yet......

Annotate return type of 
    def __enter__(self):
        return self

it it something other than ?
    def __enter__(self) -> _BitBangIO:
        return self

**Comment by tekktrik** [comment]
Resolved via #24 and #23
